### ISSUE TITLE
OpenCV doesn't seem to like bitonal files

### ISSUE THREAD

**Issue by fdncred** [Itseez/opencv#5579]
Can you please help me understand why I can't get imread to open simple bitonal files like this one?
https://app.box.com/s/y2h71ydjb7uim1um1gwqjvsuqz2cpmf2

This is an example of what I get. It worked in 3.0 rc but with 3.0.0 it no longer works.
```
Unhandled exception at 0x00007FF992BB9A1D (opencv_world300d.dll) in OpenCv_FastHough.exe: 0xC0000005: Access violation reading location 0xFFFFFFFFFFFFFFFF.
```

**Comment by StevenPuttemans** [comment]
Could you add the line of code? Also did you check latest master branch?

**Comment by fdncred** [comment]
I used the fast_hough_transform.cpp distributed in the opencv_contrib folder as a test jig.  So this is the first few lines out of the function that loads the image.
```c
static bool parseArgs(int argc, const char **argv,
    Mat &img,
    int &houghDepth,
    int &houghAngleRange,
    int &houghOperator,
    int &houghSkew)
{
    if (argc > 6)
    {
        cout << "Too many arguments" << endl;
        return false;
    }

    const char *filename = argc >= 2 ? argv[1]
        : "../../../samples/data/building.jpg";
    img = imread(filename, 0);
    if (img.empty())
    {
        cout << "Failed to load image from '" << filename << "'" << endl;
        return false;
    }
```
I have not tried the latest branch yet.  I only tried opencv 3.0rc and 3.0.0. I just did a git pull and I'm getting my dependencies in order now to compile opencv. I'll report back when I get the latest branch compiled.

**Comment by fdncred** [comment]
I've compiled the latest branch and it still fails.  I couldn't compile opencv_world so I just compiled the individual libraries to link against them.

This is the error.
```
Unhandled exception at 0x00007FF8349BE98D (opencv_imgcodecs300d.dll) in OpenCv_FastHough.exe: 0xC0000005: Access violation reading location 0xFFFFFFFFFFFFFFFF.
```
And the code breaks into the UNROLL8 line of tif_getimage.c.
```c
/*
 * 1-bit bilevel => colormap/RGB
 */
DECLAREContigPutFunc(put1bitbwtile)
{
    uint32** BWmap = img->BWmap;

    (void) x; (void) y;
    fromskew /= 8;
    while (h-- > 0) {
    uint32* bw;
    UNROLL8(w, bw = BWmap[*pp++], *cp++ = *bw++);
    cp += toskew;
    pp += fromskew;
    }
}
```

**Comment by mshabunin** [comment]
Fix from #5357 should be ported to master branch

**Comment by fdncred** [comment]
I can confirm that these changes fix my problem with the bitonal image.  Thanks.  Looking forward to the merge for this fix.  Thank you!
